makePoplogCommander.sh

```diff
@@ -639,61 +851,87 @@ cat << \****
+        } else if ( strEquals( "--run", arg0 ) ) {
+            flags = ( PREFER_SECURITY | ( flags & ~RUN_FLAGS ) );
+            deque_pop_front( argd );
+            return processArgs( argd, base, flags, envv );
+        } else if ( strEquals( "--dev", arg0 ) ) {
             //  We want to force overwrites.
-            return processOptions( argc - 1, &argv[1], base, PREFER_FLEXIBILITY, envv );
-        } else if ( strcmp( "exec", argv[1] ) == 0 ) {
-            if ( argc >= 3 ) {
-                setUpEnvironment( base, flags, envv );
-                execvp( argv[2], &argv[2] );
-            } else {
-                fprintf( stderr, "Too few arguments for exec action\n" );
-                return EXIT_FAILURE;
-            }
-        } else if ( strcmp( "shell", argv[1] ) == 0 ) {
-            char * shell_path = getenv( "SHELL" );
-            if ( shell_path == NULL ) {
-                fprintf( stderr, "$SHELL not defined\n" );
-                return EXIT_FAILURE;
+            flags = ( PREFER_FLEXIBILITY | ( flags & ~RUN_FLAGS ) );
+            deque_pop_front( argd );
+            return processArgs( argd, base, flags, envv );
+        } else if ( startsWith( arg0, "--use-build" ) ) {
+            char * build = strchr( arg0, '=' ) + 1;
+            if ( 0 ) {
+                // Never taken - a trick to regularise the following cases.
+****
+for variant in $VARIANTS
+do
+    echo "            } else if ( strEquals( build, \"${variant}\" ) ) {"
+    echo "                flags = ( VARIANT_${variant^^} | ( flags & ~VARIANT_FLAGS ) );"
+done
+cat << \****
             } else {
-                setUpEnvironment( base, flags, envv );
-                char ** shell_args = calloc( argc, sizeof( char *const ) );
-                shell_args[ 0 ] = shell_path;
-                for ( int i = 2; i < argc; i++ ) {
-                    shell_args[ i -  1 ] = argv[ i ];
-                }
-                shell_args[ argc - 1 ] = NULL; 
-                execvp( shell_path, shell_args );
+                mishap( "Unrecognised --use-build option: %s", arg0 );
             }
-        } else if ( strchr( argv[1], '=' ) != NULL ) {
-            //  If there is an '=' sign in the argument it is an environment variable.
-            chain_push( envv, argv[1] );
-            return processOptions( argc - 1, &argv[1], base, flags, envv );
+            deque_pop_front( argd );
+            return processArgs( argd, base, flags, envv );
         } else {
-            fprintf( stderr, "Unexpected arguments:" );
-            for ( int i = 1; i < argc; i++ ) {
-                fprintf( stderr, " %s", argv[ i ] );
-            }
-            fprintf( stderr, "\n" );
+            mishap( "Unrecognised --OPTION: %s", arg0 );
+        }
+    } else if ( strcmp( "exec", arg0 ) == 0 ) {
+        if ( deque_length( argd ) >= 2 ) {
+            deque_pop_front( argd );
+            setUpEnvironment( base, flags, envv );
```

Comment:
Why do we have different naming conventions here? Is there some method I'm not privy to?

Reply:
There's no good reason. The code was pulled from an old project (copyright owned by me). These days I use a cross-language capitalisation convention. We could rewrite it if it bugs you enough.